JetStreamCompose/jetstream/src/main/java/com/google/jetstream/presentation/screens/videoPlayer/components/VideoPlayerState.kt

```diff
@@ -32,18 +33,18 @@ class VideoPlayerState internal constructor(
     @IntRange(from = 0)
     val hideSeconds: Int
 ) {
-    var isDisplayed by mutableStateOf(false)
+    var controlsVisible by mutableStateOf(true)
     private val countDownTimer = MutableStateFlow(value = hideSeconds)
 
-    init {
-        MainScope().launch {
+    suspend fun observe() = coroutineScope {
+        launch {
             countDownTimer.collectLatest { time ->
                 if (time > 0) {
-                    isDisplayed = true
+                    controlsVisible = true
                     delay(1000)
                     countDownTimer.emit(countDownTimer.value - 1)
                 } else {
-                    isDisplayed = false
+                    controlsVisible = false
                 }
             }
         }
```

Comment:
I find this a bit complicated for something I think it does -- changes state after some delay

Can we simplify it to something like this:
```
class VideoPlayerState internal constructor(
    @IntRange(from = 0)
    private val hideSeconds: Int
) {
    private var _controlsVisible by mutableStateOf(true)
    val controlsVisible get() = _controlsVisible
    
    suspend fun showControls(hideAfter: Int = hideSeconds) {
        _controlsVisible = true
        delay(hideAfter.toLong())
        _controlsVisible = false
    }
}
```

In the `rememberVidePlayerState` you then just `LaunchedEffect(it) { it.showControls() }`